### PR TITLE
expression: add signature for ErrInvalidJSONData

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve? 

In the [tikv_ghpr_integration-copr-test](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_integration-copr-test/detail/tikv_ghpr_integration-copr-test/1778/pipeline) , it will appear the error ```Invalid JSON data provided to function %s: %s```. so I add the error information.

### What is changed and how it works?
Change the error information

### Check List 

Tests 
 - No code

### CODE REVIEW

**Comment by hawkingrei** [comment]
In the https://github.com/pingcap/tidb/blob/master/expression/builtin_json.go#L676, I find this error information that it will be ``` Invalid JSON data provided to function 1 : json_contains ```. 

It can be improved and will look like this.

```
Invalid JSON data provided to function json_contains  : xxxxx(Invalid JSON data)
```

File: expression/builtin_json.go [hunk 1]
```diff
@@ -1268,7 +1268,7 @@ func (b *builtinJSONKeysSig) evalJSON(row chunk.Row) (res json.BinaryJSON, isNul
 		return res, isNull, err
 	}
 	if res.TypeCode != json.TypeCodeObject {
-		return res, true, json.ErrInvalidJSONData
+		return res, true, json.ErrInvalidJSONData.GenWithStackByArgs(2, "json_contains")
```
Comment (zz-jason) [inline expression/builtin_json.go]: could you add some test for this error?

**Comment by hawkingrei** [comment]
> Thanks for your contribution, please add some tests for this.
> You can add some simple integration test in `expression/integration_test.go`:
> 
> ```
> 	tk := testkit.NewTestKit(c, s.store)
> 	err := tk.ExecToErr(`SELECT JSON_KEYS('{"a": 1, "b": {"c: 30}}')`)
> 	c.Assert(err, ...)
> ```

I find that it is not regular error and it is a bug for tidb.

Tidb:
```
mysql> SELECT JSON_KEYS('[]');
ERROR 3069 (22032): Invalid JSON data provided to function %!s(int=1): json_keys
```

Mysql
```
mysql> SELECT JSON_KEYS('[]');
+-----------------+
| JSON_KEYS('[]') |
+-----------------+
| NULL            |
+-----------------+
1 row in set (0.00 sec)
```

**Comment by hawkingrei** [comment]
@qw4990 Closed this pr. Fix in the #14556 .

**Pull request closed without merging**